src/adapter/graphql/src/scalars/pagination.rs

```diff
@@ -39,10 +39,16 @@ macro_rules! page_based_connection {
             ) -> Self {
                 let (total_pages, has_next_page) = match total_count {
                     0 => (Some(0), false),
-                    tc => (
-                        Some(tc.div_ceil(per_page)),
-                        (tc.div_ceil(per_page) - 1) > current_page,
-                    ),
+                    tc => {
+                        if per_page == 0 {
+                            (Some(0), true)
```

Comment:
My idea was that we would reach this only if `total_count` is greater than 0 and we would notify that there are more records. But I agree that often clients accept it as a flag to iterate further.
Changed

Reply:
Yeah I agree that it can be treated both ways :)

My reasoning was that if there is some silly automation that queries data until `!hasNextPage` and it somehow ends up with `perPage = 0` - to prevent it from entering an endless loop.